editor/import/resource_importer_scene.cpp

```diff
@@ -302,159 +260,88 @@ Node *ResourceImporterScene::_fix_node(Node *p_node, Node *p_root, Map<Ref<Mesh>
 		}
 	}
 
-	String name = p_node->get_name();
-
-	bool isroot = p_node == p_root;
+	const auto [node_name, node_hint] = _get_node_import_hint(p_node->get_name());
+	const bool is_root = p_node == p_root;
 
-	if (!isroot && _teststr(name, "noimp")) {
+	if (!is_root && node_hint == NODE_HINT_NO_IMPORT) {
 		memdelete(p_node);
 		return nullptr;
 	}
 
-	if (Object::cast_to<MeshInstance3D>(p_node)) {
-		MeshInstance3D *mi = Object::cast_to<MeshInstance3D>(p_node);
-
-		Ref<ArrayMesh> m = mi->get_mesh();
-
-		if (m.is_valid()) {
-			for (int i = 0; i < m->get_surface_count(); i++) {
-				Ref<StandardMaterial3D> mat = m->surface_get_material(i);
-				if (!mat.is_valid()) {
-					continue;
-				}
-
-				if (_teststr(mat->get_name(), "alpha")) {
-					mat->set_transparency(StandardMaterial3D::TRANSPARENCY_ALPHA);
-					mat->set_name(_fixstr(mat->get_name(), "alpha"));
-				}
-				if (_teststr(mat->get_name(), "vcol")) {
-					mat->set_flag(StandardMaterial3D::FLAG_ALBEDO_FROM_VERTEX_COLOR, true);
-					mat->set_flag(StandardMaterial3D::FLAG_SRGB_VERTEX_COLOR, true);
-					mat->set_name(_fixstr(mat->get_name(), "vcol"));
-				}
-			}
-		}
-
-		if (p_light_bake_mode != LIGHT_BAKE_DISABLED) {
-			mi->set_gi_mode(GeometryInstance3D::GI_MODE_BAKED);
-		}
+	if (auto *ap = Object::cast_to<AnimationPlayer>(p_node); ap) {
```

Comment:
We don't use `auto` nor in-condition-assignments in the Godot codebase.

Reply:
Hm... Before implementation I checked codebase and found `auto` usages, for example [here](https://github.com/godotengine/godot/blob/36f61030262c7661aab8407c784fd1e72eea219d/modules/gdnavigation/nav_map.cpp#L164) :(
As for me, `auto` helpes **only** in cases where type is iterator or known from the context. Here class name is already known from `cast_to`. And `auto` just helps to avoid double class typing. [clan-tidy](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html) have even a rule to use `auto` in only such cases.
But, please, let me know if you not agree with this, I will remove it of course.

This is not `in-condition-assignment`, this is an initialization inside `if` that comes from [C++17](https://en.cppreference.com/w/cpp/language/if). This feature allow to avoid polluting scope with variables that needed only inside `if` statement.
Also, let me know if you do not use it to, I will remove it too.

Reply:
> Before implementation I checked codebase and found auto usages, for example here :(

That's an oversight, or an exception due to the use of the very `std` overly verbose iterators that prompted the creation of `auto` in the first place. `auto` is banned in the Godot codebase, we like our types explicit.

Reply:
Okay, I will remove it :)

Can I leave modern C++17 initialization inside `if`? It like initialization inside `for`.

Reply:
> This is not in-condition-assignment, this is an initialization inside if that comes from C++17. This feature allow to avoid polluting scope with variables that needed only inside if statement.

We only recently started using C++17 (from a C++03 "C with classes" coding style), and we're taking a conservative approach to the use of new features.

We use the features which make obvious sense, like better templating options, guaranteed copy elision, class member initialization, etc., but for more debatable features like this one which make the code (IMO) harder to parse and more foreign to C / classical C++, I would err on the side of not using them.

It's an extremely common construct in the Godot codebase to do:
```
AnimationPlayer *ap = Object::cast_to<AnimationPlayer>(p_node);
if (ap) {

}
```

If we wanted to change this, it should be done all at once in the whole codebase (thousands of call sites) and not left to individual contributor preference.

Reply:
Not a problem, thanks for the explanations!

The last question. In one function I need to return a `Pair`. Can I use [Structured binding declaration](https://en.cppreference.com/w/cpp/language/structured_binding)?
```cpp
// With (will no work without auto)
auto [node_name, hint] = function()
static_body->set_name(node_name);

// Without
Pair<String, Hint> fixedNode = function();
static_body->set_name(fixedNode.second);
```

Reply:
As a rule of thumb, if a feature is C++17 only, you shouldn't use it unless it solves a use case that couldn't be done before C++17.

In this case an explicit `Pair<String, Hint>` is definitely what we **want**. We want to know the container we're using, and the types it's holding. `auto [a, b]` is definitely an anti-feature for us.

Reply:
Okay, thanks, will fix it today!

Reply:
Done!